### PR TITLE
Estop pub fix

### PR DESCRIPTION
**Pull request by LazyEngineerToBe**
Travail fait : Added a while(ros::ok) loop. Added the latch argument to the estop status publisher #78 

Est-ce que ça été testé et comment? : Besoin de tester sur le robot?

### CODE REVIEW

File: takin_estop/src/estop.cpp [hunk 1]
```diff
@@ -54,14 +54,19 @@ void initializeGPIO() {
...
     ros::ServiceServer serviceDisable = nh.advertiseService("takin_estop_disable", toggleEstopDisable);
-    
-    publishEstopStatus(&nh, &estop_status_pub);
-    ros::spin();
+
+    ros::Rate r(60); // 60Hz
+    while (ros::ok())
+    {
+        publishEstopStatus(&nh, &estop_status_pub);
+        // ros::spin(); // Replaced by ros::Rate
```
Comment (Ludoenso) [inline takin_estop/src/estop.cpp]: You still need to add the ros::spin()

Comment (Ludoenso) [reply takin_estop/src/estop.cpp]: Have you read the tutorial? http://wiki.ros.org/ROS/Tutorials/WritingPublisherSubscriber%28c%2B%2B%29 It explains line by line what the code does to make a publisher work.

**Comment by Ludoenso** [comment]
Why do you want to push code into master that you don't even know works? What are you trying to accomplish?

**Comment by LazyEngineerToBe** [comment]
It launches just fine on my computer, but since the node uses a library specific to the Jetson. It can't open the GPIO therefore, it can't publish anything.

**Comment by LazyEngineerToBe** [comment]
@Ludoenso Freeze this branch before testing. So that we're on the same page as to what has been tested? Please advise if there's an other way you'd like me to do that.

**Comment by Ludoenso** [comment]
> @Ludoenso Freeze this branch before testing. So that we're on the same page as to what has been tested? Please advise if there's an other way you'd like me to do that.

This branch is not frozen. The e-stop is just not tested at least on the robot. And also you're still missing code...

**Comment by Ludoenso** [comment]
> It launches just fine on my computer, but since the node uses a library specific to the Jetson. It can't open the GPIO therefore, it can't publish anything.

Right, I thought that if you cannot access the GPIO pin, you would send false. Wouldn't that be preferable?

File: takin_estop/src/estop.cpp (hunk 1 above)
Comment (LazyEngineerToBe) [reply takin_estop/src/estop.cpp]: Of course not! 🤣🤣🤣

Comment (LazyEngineerToBe) [reply takin_estop/src/estop.cpp]: I read the ros::rate one. That's what confused me.

**Pull request closed without merging**